hyperspy/signal.py

```diff
@@ -3585,19 +3585,29 @@ def _iterate_signal(self):
             yield(data[getitem])
 
     def _remove_axis(self, axis):
-        axis = self.axes_manager[axis]
-        self.axes_manager.remove(axis.index_in_axes_manager)
-        if axis.navigate is False:  # The removed axis is a signal axis
-            if self.axes_manager.signal_dimension == 2:
-                self._record_by = "image"
-            elif self.axes_manager.signal_dimension == 1:
-                self._record_by = "spectrum"
-            elif self.axes_manager.signal_dimension == 0:
-                self._record_by = ""
-            else:
-                return
-            self.metadata.Signal.record_by = self._record_by
-            self._assign_subclass()
+        am = self.axes_manager
+        axis = am[axis]
+        if am.navigation_dimension + am.signal_dimension > 1:
+            am.remove(axis.index_in_axes_manager)
+            if axis.navigate is False:  # The removed axis is a signal axis
+                if am.signal_dimension == 2:
+                    self._record_by = "image"
+                elif am.signal_dimension == 1:
+                    self._record_by = "spectrum"
+                elif am.signal_dimension == 0:
+                    self._record_by = ""
+                else:
+                    return
+                self.metadata.Signal.record_by = self._record_by
+                self._assign_subclass()
+        else:
+            # Don't remove axis because it is the only one and HyperSpy does not
+            # support 0 dimensions
+            axis.size = 1
+            axis.scale = 1
+            axis.offset = 0
+            axis.name = "scalar"
+            axis.navigate = False
```

Comment:
I agree with @vidartf . For now lets just merge the current solution so that the bug is fixed